stan/math/prim/fun/log_softmax.hpp

```diff
@@ -41,7 +41,7 @@ template <typename T, require_t<std::is_arithmetic<scalar_type_t<T>>>...>
 inline auto log_softmax(const T& x) {
   return apply_vector_unary<T>::apply(x, [&](const auto& v) {
     check_nonzero_size("log_softmax", "v", v);
-    return (v.array() - log_sum_exp(v)).matrix();
+    return (v.array() - log_sum_exp(v)).matrix().eval();
```

Comment:
Rather than adding ```.eval()``` to each of these functions, you could instead modify the ```apply_vector_unary``` header to eval the eigen returns:
```
  template <typename F>
  static inline auto apply(const T& x, const F& f) {
    return f(x).eval();
  }
```

Reply:
I went this way, so we don't need to first eval and than copy matrices in std::vector overloads that assign result of `apply_vector_unary::apply()` instead of returning it.